### PR TITLE
Spring cleaning

### PR DESCRIPTION
**Pull request by muellerzr**
Spring (winter?) cleaning thanks to [refurb](https://github.com/dosisod/refurb)

Command ran (for posterity that I can come back to next month):
```bash
refurb --ignore 109 --ignore 115 --python-version 3.7 src/ >> report.txt
```

### CODE REVIEW

File: src/accelerate/commands/launch.py [hunk 1]
```diff
@@ -98,7 +98,7 @@ def __call__(self, parser, namespace, values, option_string=None):
             "optional arguments",
         ]
         if len(args) > 1:
-            used_platforms = [arg for arg in args if arg in options_to_group.keys()]
+            used_platforms = [arg for arg in args if arg in options_to_group]
```
Comment (sgugger) [inline src/accelerate/commands/launch.py:101]: Disagree with this change here. It's more readable to explicitly have the `.keys()`.

File: src/accelerate/big_modeling.py [hunk 2]
```diff
@@ -245,7 +245,7 @@ def dispatch_model(
     check_device_map(model, device_map)
 
     if main_device is None:
-        if set(device_map.values()) == {"cpu"} or set(device_map.values()) == {"cpu", "disk"}:
+        if set(device_map.values()) in ({"cpu"}, {"cpu", "disk"}):
```
Comment (sgugger) [inline src/accelerate/big_modeling.py:248]: Not convinced this is more readable.

File: src/accelerate/commands/menu/input.py [hunk 3]
```diff
@@ -77,8 +77,7 @@ def handle_input(cls):
         if handler:
             cls.current_selection = char
             return handler(cls)
-        else:
-            return None
+        return None
```
Comment (sgugger) [inline src/accelerate/commands/menu/input.py:80]: While this is technically equivalent, I prefer the explicit `else`.

File: src/accelerate/data_loader.py [hunk 4]
```diff
@@ -164,9 +165,8 @@ def __len__(self):
         elif self.even_batches:
             # When we even batches we always get +1
             return length + 1
-        else:
-            # Otherwise it depends on the process index.
-            return length + 1 if self.process_index < len(self.batch_sampler) % self.num_processes else length
+        # Otherwise it depends on the process index.
+        return length + 1 if self.process_index < len(self.batch_sampler) % self.num_processes else length
```
Comment (sgugger) [inline src/accelerate/data_loader.py:169]: I prefer the explicit `else`.

File: src/accelerate/data_loader.py [hunk 5]
```diff
@@ -402,8 +400,7 @@ def total_batch_size(self):
     def total_dataset_length(self):
         if hasattr("total_length", self.dataset):
             return self.dataset.total_length
-        else:
-            return len(self.dataset)
+        return len(self.dataset)
```
Comment (sgugger) [inline src/accelerate/data_loader.py:403]: Same here.

File: src/accelerate/data_loader.py [hunk 6]
```diff
@@ -548,8 +543,7 @@ def __len__(self):
             return whole_length
         elif self._drop_last:
             return whole_length // self.state.num_processes
-        else:
-            return math.ceil(whole_length / self.state.num_processes)
+        return math.ceil(whole_length / self.state.num_processes)
```
Comment (sgugger) [inline src/accelerate/data_loader.py:546]: Same here.

File: src/accelerate/tracking.py [hunk 7]
```diff
@@ -500,7 +500,7 @@ def store_init_configuration(self, values: dict):
                     f'Trainer is attempting to log a value of "{value}" for key "{name}" as a parameter. MLflow\'s'
                     f" log_param() only accepts values no longer than {mlflow.utils.validation.MAX_PARAM_VAL_LENGTH} characters so we dropped this attribute."
                 )
-                del values[name]
+                values.pop(name)
```
Comment (sgugger) [inline src/accelerate/tracking.py:503]: While this is equivalent, `del values[names]` is explicit about what it's doing. `values.pop(name)` is not, so this should really be reverted.

File: src/accelerate/tracking.py [hunk 8]
```diff
@@ -593,7 +593,7 @@ def filter_trackers(
                             if getattr(tracker_init, "requires_logging_directory"):
                                 if logging_dir is None:
                                     raise ValueError(
-                                        f"Logging with `{str(log_type)}` requires a `logging_dir` to be passed in."
+                                        f"Logging with `{log_type}` requires a `logging_dir` to be passed in."
```
Comment (sgugger) [inline src/accelerate/tracking.py:596]: This is not equivalent, so this change should be reverted (f-strings use the repr, `str` does not).

File: src/accelerate/utils/megatron_lm.py [hunk 9]
```diff
@@ -538,8 +538,7 @@ def get_batch_transformer(data_iterator):
 
         if megatron_dataset_flag:
             return get_batch_megatron
-        else:
-            return get_batch_transformer
+        return get_batch_transformer
```
Comment (sgugger) [inline src/accelerate/utils/megatron_lm.py:541]: Leaving the else and return is clearer.

File: src/accelerate/utils/megatron_lm.py [hunk 10]
```diff
@@ -577,8 +576,7 @@ def loss_func_finetune(labels, logits):
 
         if pretraining_flag:
             return loss_func_pretrain
-        else:
-            return loss_func_finetune
+        return loss_func_finetune
```
Comment (sgugger) [inline src/accelerate/utils/megatron_lm.py:579]: Same here.

File: src/accelerate/utils/megatron_lm.py [hunk 11]
```diff
@@ -666,8 +664,7 @@ def get_batch_transformer(data_iterator):
 
         if megatron_dataset_flag:
             return get_batch_megatron
-        else:
-            return get_batch_transformer
+        return get_batch_transformer
```
Comment (sgugger) [inline src/accelerate/utils/megatron_lm.py:667]: Same here.

File: src/accelerate/utils/megatron_lm.py [hunk 12]
```diff
@@ -803,8 +800,7 @@ def get_batch_transformer(data_iterator):
...
```
Comment (sgugger) [inline src/accelerate/utils/megatron_lm.py:803]: Same here.

File: src/accelerate/utils/megatron_lm.py [hunk 13]
```diff
@@ -1118,8 +1114,7 @@ def eval_step(self, **batch_data):
                 else:
                     loss_reduced[key] = torch.concat(losses_reduced_for_key)
             return loss_reduced
-        else:
-            return {}
+        return {}
```
Comment (sgugger) [inline src/accelerate/utils/megatron_lm.py:1117]: Same here.

File: src/accelerate/utils/modeling.py [hunk 14]
```diff
@@ -304,7 +304,7 @@ def clean_device_map(device_map: Dict[str, Union[int, str, torch.device]], modul
     values = [v for k, v in device_map.items() if k.startswith(prefix)]
     if len(set(values)) == 1 and len(values) > 1:
         for k in [k for k in device_map if k.startswith(prefix)]:
-            del device_map[k]
+            device_map.pop(k)
```
Comment (sgugger) [inline src/accelerate/utils/modeling.py:307]: As above, this should be reverted.

File: src/accelerate/utils/operations.py [hunk 15]
```diff
@@ -225,8 +225,7 @@ def gather(tensor):
         return _gpu_gather(tensor)
     elif AcceleratorState().distributed_type == DistributedType.MULTI_CPU:
         return _cpu_gather(tensor)
-    else:
-        return tensor
+    return tensor
```
Comment (sgugger) [inline src/accelerate/utils/operations.py:228]: Let's leave the else return.

File: src/accelerate/utils/operations.py [hunk 16]
```diff
@@ -297,8 +296,7 @@ def broadcast(tensor, from_process: int = 0):
         return _gpu_broadcast(tensor, src=from_process)
     elif AcceleratorState().distributed_type == DistributedType.MULTI_CPU:
         return _gpu_broadcast(tensor, src=from_process)
-    else:
-        return tensor
+    return tensor
```
Comment (sgugger) [inline src/accelerate/utils/operations.py:299]: Same here.

File: src/accelerate/utils/operations.py [hunk 17]
```diff
@@ -439,8 +437,7 @@ def _reduce_across_processes(tensor, reduction="mean"):
         else:
             if reduction == "sum":
                 return cloned_tensor.sum()
-            else:
-                return cloned_tensor.mean()
+            return cloned_tensor.mean()
```
Comment (sgugger) [inline src/accelerate/utils/operations.py:440]: Same here.

**Review by sgugger — COMMENTED** [review]
Not really convinced by this tool, I personally disagree with many of the changes which hurt readability quite a bit.

File: src/accelerate/tracking.py (hunk 8 above)
Comment (Wauplin) [reply src/accelerate/tracking.py:596]: For the record I quickly tested it and it seems that it actually uses `__str__`, not `__rep__`. 

```py
class A:
    def __str__(self):
        return "Use str"

    def __repr__(self):
        return "Use repr"

a = A()
print(a)
print("%s" % a)
print(f"{a}")
print(f"{str(a)}")
print("{}".format(a))
```

Output (both in python 3.7 and 3.10.
```
Use str
Use str
Use str
Use str
Use str
```

Comment (sgugger) [reply src/accelerate/tracking.py]: Oh! My bad then.

Comment (Wauplin) [reply src/accelerate/tracking.py:596]: It is actually a TIL for me, didn't know what to expect :)

**Pull request merged**